subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -667,6 +671,71 @@ static void dhcpv4_timeout(struct k_work *work)
 	}
 }
 
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS_VENDOR_SPECIFIC)
+
+static int dhcpv4_parse_option_vendor(struct net_pkt *pkt, struct net_if *iface,
+				      enum net_dhcpv4_msg_type *msg_type, int length)
+{
+	struct net_dhcpv4_option_callback *cb, *tmp;
+	struct net_pkt_cursor backup;
+	uint8_t len;
+	uint8_t type;
+
+	if (length >= 3) {
+		NET_ERR("Vendor-specific option parsing, length too short");
+		net_pkt_skip(pkt, length);
```

Comment:
Why do we do this? The msg is invalid at this point.

Reply:
We are receiving some garbage from outside of the device. I don't think we can trust the message if there is some invalid data in it but should discard it.

Reply:
This was marked as resolved but IMHO it is not. If we receive garbage then we cannot continue with parsing as the data could be anything.

Reply:
This is a check, if the dhcp option 43 has encapsulated data. If the data of this option is less than 3, it can't be encapsulated, but it is still a valid option 43 send by the dhcpv4 server. Therefore we are treating it like a unknown option and skip it.

Reply:
The error print in previous line implies that there is a error in parsing, we also return a `-EBADMSG` indicating that there was an error. Why would we trust the remaining data, it seems that rest of the data could be totally bogus in this case?

Reply:
@jukkar should we return `0` instant of `-EBADMSG` in these situations?

Reply:
No, lets not change that. It is good that something is printed for the error case. I have been testing this with invalid data but have not been able to crash or get the parse to confuse itself. It does not mean that the parses works perfectly in all cases but lets deal with that when/if it happens.